packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -154,14 +159,14 @@ class ImageCache {
       _currentSizeBytes += imageSize;
       _pendingImages.remove(key);
       _cache[key] = image;
-      result.removeListener(listener);
+      result.completer.removeListener(listener);
```

Comment:
why are we using result.addListener to add the listener, but result.completer.removeListener to remove it?